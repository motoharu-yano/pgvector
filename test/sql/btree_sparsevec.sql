SET enable_seqscan = off;

CREATE TABLE t (val sparsevec(3));
INSERT INTO t (val) VALUES ('{}/3'), ('{1:1,2:2,3:3}/3'), ('{1:1,2:1,3:1}/3'), (NULL);
CREATE INDEX ON t (val);

SELECT * FROM t WHERE val = '{1:1,2:2,3:3}/3';
SELECT * FROM t ORDER BY val;

DROP TABLE t;
