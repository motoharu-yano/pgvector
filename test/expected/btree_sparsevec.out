SET enable_seqscan = off;
CREATE TABLE t (val sparsevec(3));
INSERT INTO t (val) VALUES ('{}/3'), ('{1:1,2:2,3:3}/3'), ('{1:1,2:1,3:1}/3'), (NULL);
CREATE INDEX ON t (val);
SELECT * FROM t WHERE val = '{1:1,2:2,3:3}/3';
       val       
-----------------
 {1:1,2:2,3:3}/3
(1 row)

SELECT * FROM t ORDER BY val;
       val       
-----------------
 {}/3
 {1:1,2:1,3:1}/3
 {1:1,2:2,3:3}/3
 
(4 rows)

DROP TABLE t;
